cli/src/cli-cmd-parser.c

```diff
@@ -4447,22 +4446,26 @@ cli_snap_create_parse(dict_t *dict, const char **words, int wordcount)
             goto out;
         i++;
         /* point the index to next word.
-         * As description might be follwed by force option.
+         * As description might be follwed by force option which is deprecated.
          * Before that, check if wordcount limit is reached
          */
     }
 
     if (strcmp(words[i], "force") == 0) {
-        flags = GF_CLI_FLAG_OP_FORCE;
-
+        ret = -1;
+        cli_err(
+            "\'force\' option is deprecated and"
+            "should not be used while creating snapshot");
+        gf_log("cli", GF_LOG_ERROR, "Invalid Syntax");
```

Comment:
Wouldn't it be better to show the deprecation warning but continue normally as if `force` wasn't there ? is there any functional difference that can cause unwanted results ?

Reply:
It is actually the same but without these code changes, the user would see that with `force`, the snapshot creation would go through irrespective of brick statuses ( i.e. even if a brick is down, the snapshot should be created , it is the quorum which matters ), but the actual implementation fails even if one brick is down. Now, the reasoning behind that was as glusterfs used lvm to actually do these operations, there were some issues in snapshot creation when brick is down. So, it is better if we remove the force object ( for now we can go with the deprecation warning ) but whatever comparisons or checks we have in code are actually not doing anything useful with regards to snapshot and force option.

Reply:
I still don't understand. Let's assume that a user uses `force`. With this change, `force` becomes irrelevant because we check that everything is online before doing an snapshot, right ?

In that case a simple warning saying that `force` is deprecated and the snapshot will only succeed if all bricks are online should be sufficient, and the operation will proceed. We don't need to fail the command simply because `force` is there. If a brick is down, the command will fail anyway, and that's fine, because we have warned about it.

So, if we keep the `force` option for backward compatibility, we must not fail if it's present saying that it's an invalid syntax (otherwise it's not backward compatible. It's the same as removing the option). A deprecation warning and proceeding as if it was not present seems the most optimal solution to me.